### PR TITLE
feat(driver): Abstract, Default Pipeline

### PR DESCRIPTION
**Pull request by refcell**
### Description

Abstracts the driver pipeline trait to allow for a default payload production loop.

Avoids driver implementors from footgunning themselves with rolling their own payload loops.

### CODE REVIEW

**Comment by refcell** [comment]
## 📚 $\text{Stack Overview}$

Pulls submitted in this stack:
* #796 👈
* #794
* `main`

_This comment was automatically generated by [`st`](https://github.com/clabby/st)._

File: crates/driver/src/core.rs [hunk 1]
```diff
@@ -166,7 +164,7 @@ where
             // Update the safe head.
             self.cursor.l2_safe_head = L2BlockInfo::from_block_and_genesis(
                 &block,
-                &self.pipeline.rollup_config().genesis,
+                &self.pipeline.inner().rollup_config().genesis,
```
Comment (clabby) [inline crates/driver/src/core.rs:167]: We should expose a `rollup_config` fn so we don't have to `.inner()`.

Comment (clabby) [reply crates/driver/src/core.rs:167]: same w/ `step`

**Review by clabby — COMMENTED** [review]
LGTM - to reduce the feeling of nesting, it'd be nice to lift some of the `step` / `inner` etc. functions into the pipeline itself.

**Review by clabby — APPROVED** [review]

**Pull request merged**